device-registry/src/test/scala/org/genivi/sota/device_registry/DeviceResourceSpec.scala

```diff
@@ -338,6 +340,26 @@ class DeviceResourceSpec extends ResourcePropSpec {
     }
   }
 
+  property("Can filter list of installed packages on a device") {
+    val uuid = createDeviceOk(genDeviceT.generate)
+    val pkgs = Set(PackageId(Refined.unsafeApply("foo"), Refined.unsafeApply("1.0.0")),
+      PackageId(Refined.unsafeApply("bar"), Refined.unsafeApply("1.0.0")))
+
+    installSoftware(uuid, pkgs) ~> route ~> check {
+      status shouldBe NoContent
+    }
+
+    listPackages(uuid, Some("foo")) ~> route ~> check {
+      status shouldBe OK
+      val response = responseAs[Seq[InstalledPackage]]
+      response.length shouldBe 1
+      response.head.packageId shouldEqual pkgs.head
```

Comment:
Yes, I assumed it would be ok, since it was defined, but it turns out the ordering is based on the hashes themselves.